src/services/refactors/moveToNewFile.ts

```diff
@@ -954,6 +978,22 @@ function isExported(sourceFile: SourceFile, decl: TopLevelDeclarationStatement,
         getNamesToExportInCommonJS(decl).some(name => sourceFile.symbol.exports!.has(escapeLeadingUnderscores(name)));
 }
 
+function getDefaultExportName(sourceFile: SourceFile) {
+    return tryCast((sourceFile.symbol.exports?.get("default" as __String)?.declarations?.[0] as undefined | ExportAssignment)?.expression, isIdentifier);

```

Comment:
I think you should use the function `getDefaultLikeExportNameFromDeclaration` present in `src/services/utilities.ts` here to go from the default export symbol to its name.

Reply:
It is not the same, I was using `getDefaultExportName(sourceFile)?.escapedText === name.escapedText` below to figure out wether current top-level variable declaration is exported later with default export. So `!!symbol.declarations?.some(d => isExportAssignment(d))` won't work as well as `getDefaultLikeExportNameFromDeclaration` that is used to just get name of the symbol declaration.
To better understand this check see test file: https://github.com/zardoy/typescript/blob/3295aa1d8c531f74233c9b7cb764fc1c86ebf4b0/tests/cases/fourslash/moveToNewFile_exportedLaterDefaultExports.ts